### ISSUE TITLE
Incorrect Configuration error with testIsolation setting when experimentalSessionAndOrigin is true

### ISSUE THREAD

**Issue by oatsoda** [cypress-io/cypress#24497]
### Current behavior

If you have a config like this:

```
const { defineConfig } = require("cypress");

module.exports = defineConfig({
  e2e: {
    experimentalSessionAndOrigin: true,
    testIsolation: "off",
  }
});

```

Then running Cypress _works just fine_.

*However, if you make _ANY_ change to the config, then you'll get an error:

> Your configFile at <path>cypress.config.js set an invalid value:
> 
> Expected e2e.testIsolation to be not set unless the experimentalSessionAndOrigin flag is turned on.
> 
> Instead the value was: "off"

```
Error
    at C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\packages\data-context\src\data\ProjectConfigManager.js:272:41
    at C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\packages\config\src\browser.js:121:24
    at C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\node_modules\lodash\lodash.js:4967:15
    at baseForOwn (C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\node_modules\lodash\lodash.js:3032:24)
    at C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\node_modules\lodash\lodash.js:4936:18
    at Function.forEach (C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\node_modules\lodash\lodash.js:9410:14)
    at validate (C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\packages\config\src\browser.js:111:29)
    at ProjectConfigManager.validateConfigFile (C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\packages\data-context\src\data\ProjectConfigManager.js:268:31)
    at ProjectConfigManager.initializeConfig (C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\packages\data-context\src\data\ProjectConfigManager.js:82:22)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at ProjectLifecycleManager.refreshLifecycle (C:\Users\myuser\AppData\Local\Cypress\Cache\10.11.0\Cypress\resources\app\packages\data-context\src\data\ProjectLifecycleManager.js:271:9)
```

You have to restart Cypress for it to work again.

### Desired behavior

The error is incorrect as `experimentalSessionAndOrigin` *is* `true`.  So it just shouldn't error.  

### Test code to reproduce

```
const { defineConfig } = require("cypress");

module.exports = defineConfig({
  e2e: {
    experimentalSessionAndOrigin: true,
    testIsolation: "off",
  }
});
```

### Cypress Version

10.11.0

### Node version

16.15.0

### Operating System

WIndows 11 22H2 22621.674

**Comment by mschile** [comment]
Hi @oatsoda 👋 , thanks for logging this issue! I was able to reproduce the issue and will route it to the appropriate team.

**Comment by aubincc** [comment]
Damn, I knew it wasn't me 😸

**Comment by skjnldsv** [comment]
Ahaha, same here, it drove me crazy for a couple of days 🥲

**Comment by stokrattt** [comment]
i have the same issue

**Comment by mjhenkes** [comment]
Released in [11.0.1](https://on.cypress.io/changelog#11-0-1).

This comment thread has been locked. If you are still experiencing this issue after upgrading to
Cypress v11.0.1, please open a new issue.
